src/Core/Settings.h

```diff
@@ -833,7 +833,10 @@ class IColumn;
     M(Bool, create_index_ignore_unique, false, "Ignore UNIQUE keyword in CREATE UNIQUE INDEX. Made for SQL compatibility tests.", 0) \
     M(Bool, print_pretty_type_names, false, "Print pretty type names in DESCRIBE query and toTypeName() function", 0) \
     M(Bool, create_table_empty_primary_key_by_default, false, "Allow to create *MergeTree tables with empty primary key when ORDER BY and PRIMARY KEY not specified", 0) \
-    M(Bool, allow_named_collection_override_by_default, true, "Allow named collections' fields override by default.", 0)\
+    M(Bool, allow_named_collection_override_by_default, true, "Allow named collections' fields override by default.", 0) \
+    \
+    M(UInt64, user_max_table_size_to_drop, 50000000000lu, "If size of a table is greater than this value (in bytes) than table could not be dropped with any DROP query.", 0) \
+    M(UInt64, user_max_partition_size_to_drop, 50000000000lu, "Same as max_table_size_to_drop, but for the partitions.", 0) \
```

Comment:
I was not sure about naming, but I can't use the same one we are using in server settings. 

Also, at this point, I was wondering if it makes sense to keep the server settings or if it'd be better to deprecate them and rely on these ones instead.

Reply:
@jrdi Let's name them `max_table_size_to_drop` and `max_partition_size_to_drop`.